app/utils/calculate-burn-height.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { calculateUntilBurnHeightBlockFromCycles } from './calculate-burn-height';
+
+describe(calculateUntilBurnHeightBlockFromCycles.name, () => {
+  test('it calculates correctly with one cycle', () => {
+    const result = calculateUntilBurnHeightBlockFromCycles({
+      cycles: 1,
+      rewardCycleLength: 50,
+      blocksToNextCycle: 37,
+      currentBurnBlockHeight: 1939625,
+    });
+    expect(result).toEqual(1939712);
```

Comment:
Can you elaborate? Do you mean using a real value from mainnet, rather than testnet? `674023` something? Or suggesting the maths of the fn itself should change?

Reply:
So something like:
```
genesisBurnBlockHeight + currentCycleId * rewardCycleLength + cycles * rewardCycleLength
```

Reply:
Note: I think that cycle ids are 0 indexed, not 1 indexed (the first cycle was id 0), so we could be off by one with my formula.

Reply:
Yeah, that would be give you the height of the end of the cycle, vs the height of the beginning of the cycle. 
The formula above is actually what we're using in the smart contract.